### ISSUE TITLE
How to use  both near-operation-file-preset and import-types-preset

### ISSUE THREAD

**Issue by qiwang97** [dotansimha/graphql-code-generator#3272]
I want to generate  typing declear once  in single file and other files just import its type from it .

Although `near-operation-file-preset`  `generates a file per each operation file`,every file  repeats typing declear.  And `import-types-preset `  helps import types from file but it can not generate files like  `near-operation-file-preset` .
One more thing is that `near-operation-file-preset` contains both types and components ,while splited to single file maybe better i think.  

What I want may look like this:
```
graphql
|--test1.graphql  // query and mutations
|--test1.generated.tsx  // components and hooks without types
|--test2.graphql  
|--test2.generated.tsx
|--types.ts   // types
```
But I find it's hard to work with both.

Does anyone can help me ?
Thanks a lot!

**Comment by dotansimha** [comment]
This use case is not supported yet. PRs are welcome!

**Comment by qiwang97** [comment]
> This use case is not supported yet. PRs are welcome!

sad to hear that especially a rookie like me

**Comment by karl-run** [comment]
I was exploring this as well, it turns out if you have the `typescript` plugin it generates a lot of unnecessary duplicates. I just removed the `typescript` plugin as it's not needed, and it removed over 5k lines of duplicated types, instead importing them from the shared types file.

**Comment by dotansimha** [comment]
@karl-run yeah because `typescript` generates the base types. It should be generated once into a single file. Then, each operation file should point to that file in order to load the base types from it, and choose the fields that it needs from it.

**Comment by EmirBoyaci** [comment]
> @karl-run yeah because `typescript` generates the base types. It should be generated once into a single file. Then, each operation file should point to that file in order to load the base types from it, and choose the fields that it needs from it.

Hi, how can I do that? I was exploring this but couldn't find an example to be honest.

1- I want a single file for all typescript declarations (types.ts)
2- If needed I can have operations file (operations.ts)
3- Then, I want to generate 3-4 files with typescript-react-apollo plugin based on my "documents" field in codegen config (like: user.tsx for user operations, file.tsx for file operations, etc..)

Is it possible? If yes, how? Could you please help.

Thanks,
Emir

**Comment by NilsNathorst** [comment]
Bump.

I would love to know if there has been any updates on this topic?
